core/types/types.cc

```diff
@@ -77,19 +77,21 @@ TypePtr Types::Float() {
 }
 
 TypePtr Types::arrayOfUntyped() {
-    static vector<TypePtr> targs{Types::untypedUntracked()};
+    static vector<TypePtr> targs{Types::untyped(Symbols::Magic_UntypedSource_ArrayOfUntyped())};
```

Comment:
It's likely that for these helpers, we want to take the blame as an argument, and pass in a more specific thing where they're used. For example, `hashOfUntyped` is used for both Magic buildHash as well as shapes. We likely want to bucket those separately. Similarly for the other `OfUntyped` helpers here.